ultrasync/main.py

```diff
@@ -130,24 +133,85 @@ def __init__(self, *args, **kwargs):
         # Track the Panel URL path
         self.__panel_url_path = None
 
+    def get_signature(self, date, uri, serial_number):
+        try:
+            master_key = base64.b64decode(self.master_key)
+            ascii_cs = 'US-ASCII'
+            sha256_hmac = hmac.new(master_key, msg=date.encode(ascii_cs), digestmod=hashlib.sha256)
+            sha256_hmac = hmac.new(sha256_hmac.digest(), msg=uri.encode(ascii_cs), digestmod=hashlib.sha256)
+            sha256_hmac = hmac.new(sha256_hmac.digest(), msg=serial_number.encode(ascii_cs), digestmod=hashlib.sha256)
+
+            return sha256_hmac.hexdigest()
+        except Exception as e:
+            print(e)
+            return None
+
```

Comment:
This is incredible investigative work on your part!

Reply:
As an update to this; you really did an incredible job, but I just can not accept this.  It violates GitHub's only terms and agreements.  I would not like to lose this project or any others i maintain.